johnwayodi/sm-server
# #161652102 Separate views, add more tests, Fix Sale Record issue

#### What does this PR do?
Separate the views into individual view files, fix the issue of creating and saving a sale record, add tests for edge cases
#### Description of Task to be completed?
1. Fix Sale record creation issue
2. Transfer _**SaleRecord**_ and _**SaleRecords**_  Resources to `views/sale_views.py`
3. Transfer _**Product**_ and _**ProductList**_  Resources to `views/product_views.py`
4. Transfer _**Category**_ and _**Categories**_  Resources to `views/category_views.py`
5. Transfer _**User**_ , _**UserList**_ , _**UserRegistration**_ and _**UserLogin**_  Resources to `views/user_views.py`
#### How should this be manually tested?
1. `git clone https://github.com/johnwayodi/sm-api-v2.git`
2. Ensure you are in the cloned repo, **sm-api-v2** by default.
3. Create a virtual environment using virtualenv and _pip install -r requirements.txt_
4. `git checkout -b ch-heroku-config-161652102 origin/ch-heroku-config-161652102`
5. Run `pytest -vv --cov=storemanager/api/v2` in terminal to run the tests and get coverage on the application.
6. Run the flask application and test using Postman.
#### Any background context you want to provide?
This PR is in response to implementing some of the feedback suggested.
#### What are the relevant pivotal tracker stories?
[#161615072](https://www.pivotaltracker.com/story/show/161615072)
[#161695595](https://www.pivotaltracker.com/story/show/161695595)
#### Screenshots (if appropriate)
1. Products before creation of a sale record:
![before_sale_1](https://user-images.githubusercontent.com/10683903/47955886-9cef4f00-dfaf-11e8-9cb1-d2cbf43db1be.png)

2. The Sale to be created:
![the_sale_to_make_2](https://user-images.githubusercontent.com/10683903/47955888-a7a9e400-dfaf-11e8-9aab-53b2448b5818.png)

3. Sale Record Created:
![sale_made_3](https://user-images.githubusercontent.com/10683903/47955892-be503b00-dfaf-11e8-857a-58a9e711366e.png)

4. Sale Details:
![sale_details_4](https://user-images.githubusercontent.com/10683903/47956050-df189080-dfaf-11e8-9b7d-2aa8638ee461.png)

5. Products after Sale:
![produts_after_sale_5](https://user-images.githubusercontent.com/10683903/47956056-f6577e00-dfaf-11e8-9a11-f1a064abfdad.png)

#### Questions:
Any suggestions on what to improve? leave a comment or suggestion below
